Model/Sync/CollectionsProducts/Processor.php

```diff
@@ -2,10 +2,352 @@
+     */
+    private function syncCollectionAndGetCreatedYotpoId($magentoCategoryId, $storeId)
+    {
+        $this->yotpoCategoryProcessor->processEntity($magentoCategoryId, $storeId);
+        $collectionYotpoId = $this->yotpoCategoryProcessorMain->getCategoryYotpoId($magentoCategoryId);
+        if (!$collectionYotpoId) {
+            return null;
+        }
+
+        return $collectionYotpoId;
+    }
+
+    /**
+     * @param string $magentoProductId
+     * @param string $storeId
+     * @return string|null
+     */
+    private function syncProductAndGetCreatedYotpoId($magentoProductId, $storeId)
+    {
+        $this->yotpoProductProcessorMain->setNormalSyncFlag(false);
+        $unSyncedProductIds = [$storeId => [$magentoProductId]];
+        $this->yotpoProductProcessor->process($unSyncedProductIds, [$storeId]);
+        $productYotpoId = $this->yotpoProductProcessorMain->getProductYotpoId($magentoProductId);
+        if (!$productYotpoId) {
+            return null;
+        }
+
+        return $productYotpoId;
+    }
+
+    /**
+     * @param string $storeId
+     * @param array $collectionProductEntityToSync
+     * @return void
+     */
+    private function setAsSuccessfulCollectionProductSync($storeId, $collectionProductEntityToSync)
+    {
+        $this->collectionsProductsService->updateCollectionsProductsSyncData($storeId, $collectionProductEntityToSync);
+
+        $this->catalogLogger->info(
+            __(
+                'Collections Products Sync - Synced Collection Product to Yotpo successfully - Magento Store ID: %1, Name: %2, Category ID: %3, Product ID: %4',
+                $storeId,
+                $this->coreConfig->getStoreName($storeId),
+                $collectionProductEntityToSync['magento_category_id'],
+                $collectionProductEntityToSync['magento_product_id']
+            )
+        );
+    }
+
+    /**
+     * @param string $storeId
+     * @return void
+     */
+    private function logFinishedCollectionsProductsSync($storeId)
+    {
+        $this->catalogLogger->info(
+            __(
+                'Collections Products Sync - Finished sync for store - Magento Store ID: %1, Name: %2',
+                $storeId,
+                $this->coreConfig->getStoreName($storeId)
+            )
+        );
+    }
+
+    /**
+     * @param array $collectionsProductsEntitiesToSync
+     * @param array $collectionsProductsCategoriesIds
+     * @param array $collectionsProductsProductIds
+     * @return array
+     */
+    private function enrichCollectionsProductsDataWithYotpoIds(array $collectionsProductsEntitiesToSync, array $collectionsProductsCategoriesIds, array $collectionsProductsProductIds)
+    {
+        foreach ($collectionsProductsEntitiesToSync as $collectionProductEntityToSync) {
+            $collectionsProductsCategoriesIds[] = $collectionProductEntityToSync['magento_category_id'];
+            $collectionsProductsProductIds[] = $collectionProductEntityToSync['magento_product_id'];
```

Comment:
You should use `array_unique` on both `collectionsProductsCategoriesIds` and  `collectionsProductsProductIds` - they can both contain the same ID (product or category, respectively) more than once.

Reply:
nice catch! changed it to be used with !isset? then - add to array, as `array_unique` can be heavy on performance.
cool side note, It is only used for the mysql where in query, which knows how to handle duplicates :)

Reply:
nice catch! changed it to be used with !isset? then - add to array, as I don't know if `array_unique` performance is good, we can change it to `array_unique` if we think it's better.
cool side note, this collection is used for the mysql `WHERE IN` query, which knows how to handle duplicates :).